Ensure attributeBindings work when legacy view addon is disabled.

rwjblue opened the pull request:
When the `_Ember.ENV._LEGACY_VIEW_SUPPORT` flag is falsey, the `view` stream is not setup as of 6963750. This resulted in anything added in `attributeBindings` being `undefined` (because the attribute binding system was essentially using `href=view.href` instead of `href=this.href` or just `href=href`).

You might thing (as I did initially): "HOW CAN WE NOT HAVE A SINGLE TEST THAT CHECKS FOR AN ATTRIBUTE VALUE?!?!?!?", and the answer is somewhat more disconcerting.  We currently run all tests with the legacy view and controller flags set to `true`. :(

Future test refactoring efforts will be needed to fix that mistake.

---

Fixes https://github.com/emberjs/ember.js/issues/12302.

rwjblue committed:
Ensure attributeBindings work when legacy view addon is disabled.

When the `_Ember.ENV._LEGACY_VIEW_SUPPORT` flag is falsey, the `view`
stream is not setup as of 6963750. This resulted in anything added in
`attributeBindings` being `undefined` (because the attribute binding
system was essentially using `href=view.href` instead of
`href=this.href` or just `href=href`).

You might thing (as I did initially): "HOW CAN WE NOT HAVE A SINGLE TEST
THAT CHECKS FOR AN ATTRIBUTE VALUE?!?!?!?", and the answer is somewhat
more disconcerting.  We currently run all tests with the legacy view and
controller flags set to `true`. :(

Future test refactoring efforts will be needed to fix that mistake.

nathanhammond commented:
I reviewed the stream implementation in great detail and didn't catch that this is what was resulting in the attributes not being present. We've seen similar behavior from liquid-fire–I'll confirm that this is a "likely-complete" fix tomorrow.

rwjblue committed:
Remove unsupported global check for class bindings.

rwjblue commented:
Updated to avoid using a path at all when it is not needed (`{{this.propName}}` is slightly more work than `{{propName}}`).  Also removed unsupported `isGlobal` check from `normalizeClasses`.
